runtime/service.go

```diff
@@ -192,10 +194,47 @@ func (s *service) Start(ctx context.Context, req *taskAPI.StartRequest) (*taskAP
 	if err != nil {
 		return nil, err
 	}
+	// TODO: Do we need to cancel this at some point?
+	go s.monitorState(ctx, req.ID, req.ExecID, resp.Pid)
 
 	return resp, nil
 }
 
+func (s *service) monitorState(ctx context.Context, id, exec_id string, pid uint32) {
+	ticker := time.NewTicker(time.Second)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			//make a state request
+			req := &taskAPI.StateRequest{
+				ID:     id,
+				ExecID: exec_id,
+			}
+			resp, err := s.agentClient.State(ctx, req)
+			if err != nil {
+				log.G(ctx).WithError(err).Error("error monitoring state")
+				continue
+			}
+			// if ending state, stop vm and break
+			if resp.Status == task.StatusStopped {
+				s.publish.Publish(ctx, runtime.TaskExitEventTopic, &eventstypes.TaskExit{
```

Comment:
I wonder if this is actually the right approach to take here. Polling isn't unreasonable, but shouldn't the even originate from within the VM? Furthermore, shouldn't the VM shut itself down when the task exits?

Reply:
OK